mysys/mf_iocache.cc

```diff
@@ -387,11 +387,6 @@ bool reinit_io_cache(IO_CACHE *info, enum cache_type type, my_off_t seek_offset,
   info->error = 0;
   init_functions(info);
 
-  if (info->m_encryptor != nullptr)
-    info->m_encryptor->set_stream_offset(seek_offset);
-  if (info->m_decryptor != nullptr)
-    info->m_decryptor->set_stream_offset(seek_offset);
-
```

Comment:
@kamil-holubicki What was the purpose of setting the encryptor and decryptor offset here? (in the original implementation?)

And is test case possible? pstress claims single-threaded workload will cause the bug?

Reply:
Hi @satya-bodapati 
There is no purpose in setting it here, that's why I removed it. It was added with the wrong assumption that when we reiniti IO cache, we should reinit encryptor/decryptor as well. Now (after removing this part) we are the same as upstream.

As for test, I was thinking about this, but for this we need to fill binlog cache up to particular levels. As there is no direct mapping between what queries we execute and what amount of data goes to the binlog cache I'm afraid this test will cause more maintenance pain than good.

Reply:
Ok kamil